src/Product/AttributeMappingHelper.php

```diff
@@ -105,4 +123,67 @@ public static function get_source_taxonomies(): array {
 
 		return $sources;
 	}
+
+	/**
+	 * Gets all the rules from Database
+	 *
+	 * @return array The rules from database
+	 */
+	public function get_rules(): array {
+		return $this->rules_query->get_results();
+	}
+
+	/**
+	 * Gets a specific rule from Database
+	 *
+	 * @param int $rule_id The rule ID to get from Database
+	 * @return array The rules from database
+	 */
+	public function get_rule( int $rule_id ): array {
+		return $this->rules_query->where( 'id', $rule_id )->get_row();
```

Comment:
I'm not really sure these "helper" functions should be placed in this class. Wouldn't it be better to pass an instance of `AttributeMappingRulesQuery` to the controller and move these functions within that class.

Also in this case these functions have a very specific name, but a lot of this is functionality that can be reused for any table with an ID column. Should they be part of the `Query` class?
Or if we don't want them as part of the `Query` class they could be added as a trait so they can be reused for any table with an ID column (which is currently all our tables).

Something like:
```php
function get_row_by_id( int $id ): array;
function insert_and_get_row( array $data ): ?array;
function update_and_get_row( array $data, int $id ): ?array;
function delete_by_id( int $id ): bool;
```

Reply:
Ok! 

Implemented here https://github.com/woocommerce/google-listings-and-ads/pull/1687/commits/b44b973913631ab341ee22abe1c9335c8fbf5276